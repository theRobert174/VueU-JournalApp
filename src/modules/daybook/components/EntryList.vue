<template>
    <div class="entry-list-container">
        <div class="px-2 pt-2">
            <input type="text" name="" id="" class="form-control" placeholder="Buscar Entrada">
        </div>
        <div class="entry-scrollarea">
            <Entry v-for="item in 100" :key="item"/>
        </div>
    </div>
</template>

<script>
import { defineAsyncComponent } from 'vue';

export default {
    components: {
        Entry: defineAsyncComponent( () => import('@/modules/daybook/components/EntryComponent.vue'))
    }
}
</script>

<style lang="scss" scoped>
.entry-list-container{
    border-right: 1px solid #2c3e50;
    height: calc(100vh - 56px);
}
.entry-scrollarea{
    height: calc(100vh - 100px);
    overflow: scroll;
    overflow-x: hidden;
}
</style>