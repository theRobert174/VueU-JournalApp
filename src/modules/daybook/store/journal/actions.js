import journalApi from "@/api/journalApi"

// export const myAction = async({commit}) => {


// }

export const loadEntries = async({commit}) => {
    const {data} = await journalApi.get('/entries.json')
    const entries = []
    for(let id of Object.keys(data)){
        entries.push({
            id,
            ...data[id]
        })
    }
    commit('setEntries', entries)
}
export const updateEntry = async({commit},entry) => {
    // console.log(entry, 'action')
    const {id, ...rest} = entry
    //console.log(rest)

    const resp = await journalApi.put(`/entries/${id}.json`, rest)
    console.log('resp',resp)
    commit('updateEntry', {...entry})
}
export const createEntry = async(/*{commit}*/) => {

}
export const deleteEntry = async(/*{commit}*/) => {
    
}