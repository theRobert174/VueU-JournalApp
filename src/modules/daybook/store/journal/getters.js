
// export const myGetter = (state) => {
//     return state.algo
// }
export const getEntriesByTerm = (/*state*/) => {
    //return state.algo
}
export const getEntryById = (/*state*/) => {
    //return state.algo
}