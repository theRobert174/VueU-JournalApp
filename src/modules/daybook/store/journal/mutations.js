
// export const myMutation = (state) => {

// }
export const setEntries = (state, entries) => {
    state.entries = [...state.entries, ...entries]
    console.log(entries)
    state.isLoading = false
}
export const updateEntries = (/*state*/) => {

}
export const addEntries = (/*state*/) => {

}