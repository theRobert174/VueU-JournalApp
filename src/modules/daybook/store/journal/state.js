

export default () => ({
    isLoading: true,
    entries: [
        {
            id: new Date().getTime(), //12354988461
            date: new Date().toDateString(),// sat 23, julio
            text: 'Aliquip ad reprehenderit ad incididunt elit aliqua. Eiusmod proident adipisicing exercitation cupidatat et. Cillum anim ipsum deserunt voluptate eiusmod esse do ea sunt nisi aliqua esse reprehenderit sit. In non amet aliquip sint. Esse deserunt tempor consequat magna labore occaecat Lorem voluptate consectetur ipsum id proident incididunt. Elit cupidatat nulla cillum tempor duis ut incididunt tempor exercitation velit minim.',
            picture: null, //https://
        },
        {
            id: new Date().getTime() + 1000, //12354988461
            date: new Date().toDateString(),// sat 23, julio
            text: 'Adipisicing nostrud tempor est mollit est officia veniam irure veniam id officia. Non amet cupidatat et in commodo nisi veniam ut minim tempor. Officia proident occaecat deserunt est ad tempor esse ea fugiat voluptate ut exercitation est duis.',
            picture: null, //https://
        },
        {
            id: new Date().getTime() + 2000, //12354988461
            date: new Date().toDateString(),// sat 23, julio
            text: 'Tempor eu aliquip occaecat anim occaecat ipsum dolore sint laboris ut deserunt proident. Ad est adipisicing consequat cupidatat ipsum nisi. Officia anim irure irure sit ullamco ea eiusmod irure reprehenderit. Eiusmod deserunt reprehenderit commodo aliqua aliqua cupidatat magna do fugiat ad adipisicing est. Deserunt reprehenderit amet aute exercitation aute consectetur.',
            picture: null, //https://
        },
    ]
})