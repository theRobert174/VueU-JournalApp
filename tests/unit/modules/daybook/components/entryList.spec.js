import { shallowMount } from "@vue/test-utils";
import { createStore } from "vuex";

import journal from "@/modules/daybook/store/journal"
import EntryList from "@/modules/daybook/components/EntryList.vue";

import { journalState } from "../../../mock-data/test-journal-state";


const createVuexStore = (initialState) => createStore({
    modules:{
        journal: {
            ...journal,
            state: {...initialState}
        }
    }
})

describe('Pruebas para EntryList', () => {    
    
    const store = createVuexStore(journalState)
    const mockRouter = {
        push: jest.fn()
    }

    let wrapper

    beforeEach(() => {
        jest.clearAllMocks()
        wrapper = shallowMount(EntryList, {
            global: {
                mocks: {
                    $router: mockRouter
                },
                plugins:[ store ]
            }
        })
    })

    test('debe de llamar el getEntriesByTerm sin termino y mostrar dos entradas', () => {
        expect(wrapper.findAll('entry-stub').length).toBe(2)
        expect(wrapper.html()).toMatchSnapshot()
    })

    test('debe llamar el gerEntriesByTerm y filtrar las entradas', async() => {
        const input = wrapper.find('input')
        await input.setValue('Vue')

        expect(wrapper.findAll('entry-stub').length).toBe(1)
    })

    test('el boton de nuevo debe de redireccionar a /new', () => {
        wrapper.find('button').trigger('click')

        expect(mockRouter.push).toHaveBeenLastCalledWith({name: 'entry', params: {id: 'new'}})
    })
})