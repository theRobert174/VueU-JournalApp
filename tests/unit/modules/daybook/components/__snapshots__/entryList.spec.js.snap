// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas para EntryList debe de llamar el getEntriesByTerm sin termino y mostrar dos entradas 1`] = `
<div class="entry-list-container">
  <div class="px-2 pt-2"><input type="text" name="" id="" class="form-control" placeholder="Buscar Entrada"></div>
  <div class="mt-2 d-flex flex-column"><button class="btn btn-primary mx-3"><i class="fa fa-plus-circle"></i> Nueva entrada </button></div>
  <div class="entry-scrollarea">
    <entry-stub entry="[object Object]"></entry-stub>
    <entry-stub entry="[object Object]"></entry-stub>
  </div>
</div>
`;
